src/javascript-wrapper.js

```diff
@@ -0,0 +1,188 @@
+ */
+module.exports = function( nodeDefOrUpdater ) { 
+
+    // Variables needed to create our pipeline component
+    var updaterArgs;
+    var updater;
+    var nodeDef; 
+
+    // Do we have an updater specified?  If not, use the default one
+    if (_.isUndefined( nodeDefOrUpdater ) || _.isEmpty( nodeDefOrUpdater )) { 
+        updater  = defaultUpdater;
+    }
+
+    if ( _.isFunction( nodeDefOrUpdater ) ) { 
+
+        updater = nodeDefOrUpdater;
+        updaterArgs = introspect( updater ); // get an array of updater parameter names in order
+        // Convert array to a hash of ports with data type all
+        var inPorts = ( _.isEmpty( updaterArgs ) ) ? { input: { datatype: 'all', required: true } } :
+                      updaterArgs.reduce(function(map, obj) {
+                          map[obj] = { datatype: 'all' };
+                          return map;
+                      }, {});
+        nodeDef = { inPorts: inPorts,
+                    updater: updater };
+
+    } else { 
+
+       // Have a classical component node definition - use it 
+       nodeDef = nodeDefOrUpdater || {};
+
+       // use default updater if we do not have one or nodeDef.updater if we do
+       if ( _.isUndefined( nodeDef.updater ) || ! _.isFunction( nodeDef.updater )) { 
+           updater = defaultUpdater;
+       } else { 
+	   updater = nodeDef.updater;
+       }
+       updaterArgs = introspect( updater );
+
+       // Now merge the nodeDef inPorts definition with whatever is in our updater parameters
+       // if there are any differences so we have all ports the updater may need from both 
+       // the node definition and the updater API
+
+       // Get node inPorts as a hash if it's not already
+       var inPorts = ( _.isEmpty( nodeDef.inPorts ) ) ? {} : 
+                               ( _.isArray( nodeDef.inPorts )) ? 
+                                   nodeDef.inPorts.reduce( function( map, obj ) { 
+                                       if ( _.isObject( obj ) ) { 
+                                           return obj;
+                                       }
+                                       return { [obj]: {dataType: 'all'} };
+                                    }, {})
+                                  : nodeDef.inPorts;
+       var inPortNames = Object.keys( inPorts );
+       var undefinedInPortNames  = _.difference( updaterArgs, inPortNames );
+       nodeDef.inPorts =  (_.isEmpty( undefinedInPortNames )) ? inPorts : 
+                               _.defaults( inPorts,
+                                           undefinedInPortNames.reduce(function(map, obj) {
+                                               map[obj] = { datatype: 'all' };
+                                               return map;
+					       }, {}));
+    }
+
+    // TODO: Add additional wrapper functions to this object
+    var wrapper = {fRunUpdater: fRunUpdater.bind(this, updater, updaterArgs)};
```

Comment:
As commented above, I suggest eliminating 'this' (because it isn't used) and changing bind to _.partial:
<pre>
var wrapper = {fRunUpdater: _.partial(fRunUpdater, updater, updaterArgs)};
</pre>